subcommand/lifecycle-sidecar/command_test.go

```diff
@@ -84,6 +89,115 @@ func TestRun_ConsulBinaryMissing(t *testing.T) {
 	require.Contains(t, ui.ErrorWriter.String(), "-consul-binary \"/not/a/valid/path\" not found")
 }
 
+// Test that we register the services.
+func TestRun_ServicesRegistration(t *testing.T) {
+	t.Parallel()
+	tmpDir, err := ioutil.TempDir("", "")
+	require.NoError(t, err)
+	defer func() { os.RemoveAll(tmpDir) }()
+
+	configFile := filepath.Join(tmpDir, "svc.hcl")
+	err = ioutil.WriteFile(configFile, []byte(servicesRegistration), 0600)
+	require.NoError(t, err)
+
+	a := agent.NewTestAgent(t, t.Name(), `primary_datacenter = "dc1"`)
```

Comment:
Using the TestAgent means we won't be able to add any namespace tests to the sidecar, which it would be good to do. What about using the Consul test server in the sdk instead?

Reply:
Yep, good point. I think I assumed that for namespaces we'd need to create a new test file anyways and would deal with it there. I've updated it.